modules/phase_field/src/action/MultiAuxVariablesAction.C

```diff
@@ -0,0 +1,73 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "MultiAuxVariablesAction.h"
+#include "Factory.h"
+#include "Parser.h"
+#include "FEProblem.h"
+#include "AddVariableAction.h"
+#include "Conversion.h"
+
+#include <sstream>
+#include <stdexcept>
+
+// libMesh includes
+#include "libmesh/libmesh.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/nonlinear_implicit_system.h"
+#include "libmesh/explicit_system.h"
+#include "libmesh/string_to_enum.h"
+
+const Real MultiAuxVariablesAction::_abs_zero_tol = 1e-12;
+
+template<>
+InputParameters validParams<MultiAuxVariablesAction>()
+{
+  MooseEnum familyEnum = AddVariableAction::getNonlinearVariableFamilies();
+  MooseEnum orderEnum = AddVariableAction::getNonlinearVariableOrders();
+
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Set up auxvariables for a polycrystal sample");
+  params.addParam<MooseEnum>("family", familyEnum, "Specifies the family of FE shape functions to use for this variable");
+  params.addParam<MooseEnum>("order", orderEnum, "Specifies the order of the FE shape function to use for this variable");
```

Comment:
Yes, look at the logic in `AddAuxVariableAction::act()`. You need to utilize the `blocks` input parameter and pass it to `_problem->addAuxVariable` if it is not empty.